### PR TITLE
test: add test to confirm "next error code" comment is updated

### PR DESCRIPTION
**Pull request by nolanlawson**
## Details

As discussed here: https://github.com/salesforce/lwc/pull/2372#discussion_r652445115

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

### CODE REVIEW

File: packages/@lwc/errors/src/__tests__/errors.spec.ts [hunk 1]
```diff
@@ -75,4 +77,22 @@ describe('error validation', () => {
 
         traverseErrorInfo(CompilerErrors, checkUniqueness, 'compiler');
     });
+
+    it('Next error code is updated', () => {
```
Comment (jodarove) [inline packages/@lwc/errors/src/__tests__/errors.spec.ts:81]: should we add info about where to add the new error code (`../compiler/error-info/index.ts`)?

File: packages/@lwc/errors/src/compiler/error-info/index.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@
  */
 /**
  * TODO [W-5678919]: implement script to determine the next available error code
```
Comment (jodarove) [inline packages/@lwc/errors/src/compiler/error-info/index.ts:8]:
```suggestion
```

File: packages/@lwc/errors/src/__tests__/errors.spec.ts [hunk 3]
```diff
@@ -75,4 +77,22 @@ describe('error validation', () => {
...
+        const errorCodes: number[] = [];
+        traverseErrorInfo(
+            CompilerErrors,
+            (error) => {
+                errorCodes.push(error.code);
+            },
+            'compiler'
+        );
+        const expectedNextErrorCode = 1 + Math.max(...errorCodes);
```
Comment (jodarove) [inline packages/@lwc/errors/src/__tests__/errors.spec.ts:90]: a different approach to achieve this, may be (either way is fine):

```suggestion
        let lastErrorCode = 1001;
        traverseErrorInfo(
            CompilerErrors,
            (error) => {
                lastErrorCode = Math.max(lastErrorCode, error.code);
            },
            'compiler'
        );
        const expectedNextErrorCode = 1 + lastErrorCode;
```

**Review by jodarove — APPROVED** [review]
lgtm, just some minor suggestions.

File: packages/@lwc/errors/src/__tests__/errors.spec.ts (hunk 1 above)
Comment (nolanlawson) [reply packages/@lwc/errors/src/__tests__/errors.spec.ts:81]: done

File: packages/@lwc/errors/src/__tests__/errors.spec.ts (hunk 3 above)
Comment (nolanlawson) [reply packages/@lwc/errors/src/__tests__/errors.spec.ts:90]: Sure, this is fine!

**Review by jodarove — APPROVED** [review]
lgtm

**Review by pmdartus — APPROVED** [review]

File: packages/@lwc/errors/src/__tests__/errors.spec.ts [hunk 4]
```diff
@@ -75,4 +77,22 @@ describe('error validation', () => {
 
         traverseErrorInfo(CompilerErrors, checkUniqueness, 'compiler');
     });
+
+    it('Next error code is updated in src/compiler/error-info/index.ts', () => {
```
Comment (ekashida) [inline packages/@lwc/errors/src/__tests__/errors.spec.ts:81]: Nit: If someone updates the error code without seeing the comment to being with, they might be confused without a little more context.
```suggestion
    it('"Next error code: XXXX" comment is updated in src/compiler/error-info/index.ts', () => {
```

**Review by ekashida — APPROVED** [review]

**Review by ravijayaramappa — APPROVED** [review]

**Pull request merged**